src/catalog/model/payment/paymentwall.php

```diff
@@ -71,4 +72,55 @@ private function prepareDeliveryData($order, $ref)
             'is_test' => $this->config->get('paymentwall_test') ? 1 : 0,
         );
     }
+
+    public function generateWidget($orderInfo, $customer, $successUrl)
+    {
+        // Init Paymentwall configs
+        $this->initConfig();
+
+        $successUrl = $this->config->get('paymentwall_success_url')
+                    ? $this->config->get('paymentwall_success_url')
+                    : $successUrl;
+
+        $widget = new Paymentwall_Widget(
+            !empty($customer->getId()) ? $customer->getId() : $_SERVER["REMOTE_ADDR"],
+            $this->config->get('paymentwall_widget'),
+            array(
+                new Paymentwall_Product(
+                    $orderInfo['order_id'],
+                    $orderInfo['currency_value'] > 0 ? $orderInfo['total'] * $orderInfo['currency_value'] : $orderInfo['total'], // when currency_value <= 0 changes to 1
+                    $orderInfo['currency_code'],
+                    'Order #' . $orderInfo['order_id']
+                )
+            ),
+            array_merge(
+                array(
+                    'success_url' => $successUrl,
+                    'integration_module' => 'opencart',
+                    'test_mode' => $this->config->get('paymentwall_test')
+                ),
+                $this->getUserProfileData($orderInfo)
+            ));
+
+        return $widget->getHtmlCode(array(
+            'width' => '100%',
+            'height' => 400,

```

Comment:
let's increase height to `600`
I think we should have some options to update style for iframe, it will looks better